doc/releases/changelog-0.38.0.md

```diff
@@ -4,112 +4,396 @@
+      Wires([1, 4])
+      ```
+
+    * union with `|` or `union()`:
+
+      ```python
+      >>> wires1.union(wires2) # or wires1 | wires2
+      Wires([1, 2, 3, 4])
+      ```
+
+    * difference with `-` or `difference()`:
+
+      ```python
+      >>> wires1.difference(wires2) # or wires1 - wires2
+      Wires([1])
+      ```
 
 <h4>Quantum arithmetic operations 🧮</h4>
 
-* The `qml.Adder` and `qml.PhaseAdder` templates are added to perform in-place modular addition.
+* Several new operator templates have been added to PennyLane that let you perform quantum arithmetic 
+  operations.
   [(#6109)](https://github.com/PennyLaneAI/pennylane/pull/6109)
-
-* The `qml.Multiplier` and `qml.OutMultiplier` templates are added to perform modular multiplication.
   [(#6112)](https://github.com/PennyLaneAI/pennylane/pull/6112)
-
-* The `qml.OutAdder` and `qml.ModExp` templates are added to perform out-of-place modular addition and modular exponentiation.
   [(#6121)](https://github.com/PennyLaneAI/pennylane/pull/6121)
 
+  * `qml.Adder` performs in-place modular addition: 
+    :math:`\text{Adder}(k, m)\vert x \rangle = \vert x + k \; \text{mod} \; m\rangle`. 
 
-<h4>Creating spin Hamiltonians 🧑‍🎨</h4>
+  * `qml.PhaseAdder` is similar to `qml.Adder`, but it performs in-place modular addition in the Fourier 
+    basis. 
 
-* The function ``transverse_ising`` is added to generate transverse-field Ising Hamiltonian.
-  [(#6106)](https://github.com/PennyLaneAI/pennylane/pull/6106)
+  * `qml.Multiplier` performs in-place multiplication: 
+    :math:`\text{Multiplier}(k, m)\vert x \rangle = \vert x \times k \; \text{mod} \; m \rangle`.
 
-* The functions ``heisenberg`` and ``fermi_hubbard`` are added to generate Heisenberg and Fermi-Hubbard Hamiltonians respectively.
-  [(#6128)](https://github.com/PennyLaneAI/pennylane/pull/6128)
+  * `qml.OutAdder` performs out-place modular addition:
+    :math:`\text{OutAdder}(m)\vert x \rangle \vert y \rangle \vert b \rangle = \vert x \rangle \vert y \rangle \vert b + x + y \; \text{mod} \; m \rangle`.
+
+  * `qml.OutMultiplier` performs out-place modular multiplication: 
+    :math:`\text{OutMultiplier}(m)\vert x \rangle \vert y \rangle \vert b \rangle = \vert x \rangle \vert y \rangle \vert b + x \times y \; \text{mod} \; m \rangle`.
+
+  * `qml.ModExp` performs modular exponentiation: 
+    :math:`\text{ModExp}(base, m) \vert x \rangle \vert k \rangle = \vert x \rangle \vert k \times base^x \; \text{mod} \; m \rangle`.
+
+  Here is a comprehensive example that performs the following calculation: `(2 + 1) * 3 mod 7 = 2` (or 
+  `010` in binary).
+
+  ```python
+  dev = qml.device("default.qubit", shots=1)
+
+  wire_reg = qml.registers({
+      "x_wires": 2, # |x>: stores the result of 2 + 1 = 3
+      "y_wires": 2, # |y>: multiples x by 3
+      "output_wires": 3, # stores the result of (2 + 1) * 3 m 7 = 2
+      "work_wires": 2 # for qml.OutMultiplier
+  })
```

Comment:
Nice to use registers here! But I get if we don't want to use it in this setting because it might take away / be distracting from the arithmetic features

Reply:
As a reader, it took me a while to parse the use of registers for different parts of `(2 + 1) * 3 mod 7 = 2`. But I think it would be even harder with lists of wires :thinking: